packages/flame_tiled/lib/src/renderable_layers/tile_layer.dart

```diff
@@ -0,0 +1,460 @@
+        final flips = SimpleFlips.fromFlips(tileGid.flips);
+        final scale = size.x / src.width;
+        final anchorX = src.width - halfMapTile.x;
+        final anchorY = src.height - halfMapTile.y;
+
+        late double offsetX;
+        late double offsetY;
+
+        // halfTile.x: shifts the map half a tile forward rather than
+        //             lining up on at the center.
+        // halfTile.y: shifts the map half a tile down rather than
+        //             lining up on at the center.
+        // StaggerX/Y: Moves the tile forward/down depending on orientation.
+        //  * stagger: Hexagonal tiles move down or right by only a fraction,
+        //             specifically 3/4 the width or height, for packing.
+        if (_map.staggerAxis == StaggerAxis.y) {
+          offsetX = tx * size.x + staggerX + halfDestinationTile.x;
+          offsetY = ty * staggerY + halfDestinationTile.y;
+        } else {
+          offsetX = tx * staggerX + halfDestinationTile.x;
+          offsetY = ty * size.y + staggerY + halfDestinationTile.y;
+        }
+
+        final scos = flips.cos * scale;
+        final ssin = flips.sin * scale;
+
+        final transform = ui.RSTransform(
+          scos,
+          ssin,
+          offsetX + -scos * anchorX + ssin * anchorY,
+          offsetY + -ssin * anchorX - scos * anchorY,
+        );
+
+        // A second pass is only needed in the case of staggery.
+        if (_map.staggerAxis == StaggerAxis.x && staggerY > 0) {
+          xSecondPass.add(_Transform(src, transform, flips.flip, batch));
+        } else {
+          batch.addTransform(
+            source: src,
+            transform: transform,
+            flip: flips.flip,
+          );
+        }
+      }
+
+      for (final tile in xSecondPass) {
+        tile.batch.addTransform(
+          source: tile.source,
+          transform: tile.transform,
+          flip: tile.flip,
+        );
+      }
+    }
+  }
+
+  @override
+  void render(Canvas canvas, Camera? camera) {
+    canvas.save();
+
+    canvas.translate(offsetX, offsetY);
+
+    if (camera != null) {
+      _applyParallaxOffset(canvas, camera);
+    }
+
+    for (final batch in _cachedSpriteBatches.values) {
+      batch.render(canvas, paint: _layerPaint);
+    }
+
+    canvas.restore();
+  }
+
+  static Future<_RenderableLayer> load(
+    TileLayer layer,
+    _GroupLayer? parent,
+    TiledMap map,
+    Vector2 destTileSize,
+  ) async {
+    return _TileLayer(
+      layer,
+      parent,
+      map,
+      destTileSize,
+      await _loadImages(map),
+    );
+  }
+
+  static Future<Map<String, SpriteBatch>> _loadImages(TiledMap map) async {
+    final result = <String, SpriteBatch>{};
+
+    for (final img in map.tiledImages()) {
+      final src = img.source;
+      if (src != null) {
+        result[src] = await SpriteBatch.load(src);
+      }
+    }
+
+    return result;
+  }
+}
+
+/// Caches transforms for staggered maps as the row/col are switched.
```

Comment:
Can do. Shall I call this new file simply `transform.dart`?

Reply:
follows the theme - I'm sold.  I have new things in the pipeline that I'll make into new files as well (like stack, and animations)

Reply:
I wonder if it would be better to call it `TileTransform` or something more specific so that it doesn't get confused with any of the other transform classes.

Reply:
Oh, in that case, should we create a new directory called `utils` (or something similar) for such auxiliary files?

Reply:
> I wonder if it would be better to call it `TileTransform` or something more specific so that it doesn't get confused with any of the other transform classes.

I'm fine with `_TileTransform` since its very specific.

Reply:
> Oh, in that case, should we create a new directory called `utils` (or something similar) for such auxiliary files?

`src/utils`?

Reply:
> `src/utils`?

Yeah

Reply:
👍🏻 - I think we're racing to see who's PR lands first 🏎️

Reply:
We try to stay away from `utils` directories as much as we can, because it usually ends up being disorganized with several different unrelated things in the same directory. Since this package still is quite small, can't it just live directly in `src`?

Reply:
> Since this package still is quite small, can't it just live directly in `src`?

Oops, just read your comment.